### PR TITLE
Add menu bar TS examples

### PR DESCRIPTION
**Pull request by yuriy-fix**

### CODE REVIEW

File: frontend/demo/component/menubar/menu-bar-styles.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      }
+    `;
+  }
+
+  render() {
+    return html`
+      <!-- tag::snippet[] -->
+      <vaadin-menu-bar .items="${[{ text: 'Default', children: [{ text: 'Item' }]}]}"></vaadin-menu-bar>
+      <vaadin-menu-bar theme="tertiary" .items="${[{ text: 'Tertiary', children: [{ text: 'Item' }]}]}"></vaadin-menu-bar>
+      <vaadin-menu-bar theme="primary" .items="${[{ text: 'Primary', children: [{ text: 'Item' }]}]}"></vaadin-menu-bar>
```
Comment (yuriy-fix) [inline frontend/demo/component/menubar/menu-bar-styles.ts:30]: The corners are not rounded, it's really visible in this example. Issue created for this one: https://github.com/vaadin/web-components/issues/278

File: frontend/demo/component/menubar/menu-bar-icon-only.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      <!-- tag::snippethtml[] -->
+      <vaadin-menu-bar theme="tertiary-inline" .items="${this.items}"></vaadin-menu-bar>
+      <!-- end::snippethtml[] -->
+    `;
+  }
+
+  createItem(iconName: string, ariaLabel: string) {
+    const item = window.document.createElement('vaadin-context-menu-item');
+    const icon = window.document.createElement('iron-icon');
+    item.setAttribute('aria-label', ariaLabel);
```
Comment (yuriy-fix) [inline frontend/demo/component/menubar/menu-bar-icon-only.ts:60]: The `aria-label` is set on the item, but not on the wrapping button. Probably should be fixed within https://github.com/vaadin/web-components/issues/277.

File: articles/ds/components/menu-bar/index.asciidoc [hunk 3]
```diff
@@ -11,31 +11,320 @@ section-nav: incomplete
 = Menu Bar
 
 // tag::description[]
-Menu Bar is a component for displaying a horizontal menu with multi-level sub-menus.
+The Menu Bar is a horizontal button bar with hierarchical drop-down menus.
```
Comment (magi42) [inline articles/ds/components/menu-bar/index.asciidoc:14]:
```suggestion
Menu Bar is a horizontal button bar with hierarchical drop-down menus.
```
I think we have agreed not to use article with component names.

File: articles/ds/components/menu-bar/index.asciidoc [hunk 4]
```diff
@@ -11,31 +11,320 @@ section-nav: incomplete
...
+Menu items can either trigger an action, open a menu or work as a toggle.
```
Comment (magi42) [inline articles/ds/components/menu-bar/index.asciidoc:15]:
```suggestion
Menu items can either trigger an action, open a menu, or work as a toggle.
```

**Review by magi42 — COMMENTED** [review]
Just couple of comments about the text, I didn't check the example code itself.

File: articles/ds/components/menu-bar/index.asciidoc [hunk 5]
```diff
@@ -11,31 +11,320 @@ section-nav: incomplete
...
+// [source,java]
+// ----
+// include::../../../../src/main/java/com/vaadin/demo/component/menubar/MenuBarComboButtons.java[render,tags=snippet,indent=0,group=Java]
+// ----
+
+--
+
+== Best Practices
+
+* Menu Bar should not be used for navigation. Use <<../tabs#,Tabs>> for switching between content, or <<../anchor#,Anchors>> for regular navigation.
```
Comment (sosa-vaadin) [inline articles/ds/components/menu-bar/index.asciidoc:311]:
```suggestion
* Menu Bar should not be used for navigation. Use <<../tabs#,tabs>> for switching between content, or <<../anchor#,anchors>> for regular navigation.
```

File: articles/ds/components/menu-bar/index.asciidoc [hunk 6]
```diff
@@ -11,31 +11,320 @@ section-nav: incomplete
...
+* Menu Bar is not an input field. Use <<../select#,Select>>, <<../combo-box#,Combo Box>>, or <<../radio-button#,Radio Buttons>> instead.
```
Comment (sosa-vaadin) [inline articles/ds/components/menu-bar/index.asciidoc:312]:
```suggestion
* Menu Bar is not an input field. Use <<../select#,Select>>, <<../combo-box#,Combo Box>>, or <<../radio-button#,Radio Button>> instead.
```

File: frontend/demo/component/menubar/menu-bar-combo-buttons.ts [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+      component: this.createItem(),
+      children: [{ text: 'Save as draft' }, { text: 'Save as copy' }, { text: 'Save and publish' }],
+    },
+  ];
+  // end::snippet[]
+
+  render() {
+    return html`
+      <!-- tag::snippethtml[] -->
+      <vaadin-menu-bar theme="primary" .items="${this.items}"></vaadin-menu-bar>
```
Comment (sosa-vaadin) [inline frontend/demo/component/menubar/menu-bar-combo-buttons.ts:32]:
```suggestion
      <vaadin-menu-bar theme="primary icon" .items="${this.items}"></vaadin-menu-bar>
```
Maybe adding an `icon` theme will make the buttons smaller and better looking.

**Review by sosa-vaadin — COMMENTED** [review]
I left some small suggestions. I will approve once they're answered or implemented.

File: frontend/demo/component/menubar/menu-bar-checkable.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+        @item-selected="${this.itemSelected}"
+      ></vaadin-menu-bar>
+      <!-- end::snippethtml[] -->
+    `;
+  }
+
+  // tag::snippetselected[]
+  itemSelected(e: MenuBarItemSelectedEvent) {
+    const item = e.detail.value;
+    (item as any).checked = !(item as any).checked;
```
[the remark this answers is not in the document]
Comment (yuriy-fix) [reply frontend/demo/component/menubar/menu-bar-checkable.ts]: Using `SubMenuItem` here.

File: frontend/demo/component/menubar/menu-bar-combo-buttons.ts (hunk 7 above)
Comment (yuriy-fix) [reply frontend/demo/component/menubar/menu-bar-combo-buttons.ts]: I was thinking about that, but that one is not a part of `menu-bar`, but getting it from `button` themes.
Not sure if we should add it here, but I guess it's not a problem 🤔

**Review by sosa-vaadin — APPROVED** [review]

File: frontend/demo/component/menubar/menu-bar-combo-buttons.ts (hunk 7 above)
Comment (jouni) [reply frontend/demo/component/menubar/menu-bar-combo-buttons.ts]: I don’t see a problem. Menu Bar uses the button variants, "primary" is also coming from button I believe, not redefined in Menu Bar.

**Comment by tomivirkki** [comment]
Screenshot 2021-05-05 at 10 09 50

This looks messy.

**Pull request merged**